### PR TITLE
storage: Btrfs improvements

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Review by jelly — APPROVED** [review]

**Comment by mvollmer** [comment]
Had to update the p-tests.

File: pkg/storaged/block/format-dialog.jsx [hunk 1]
```diff
@@ -290,7 +290,7 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
     let [, old_dir, old_opts] = get_fstab_config(block, true,
                                                  content_block?.IdType == "btrfs"
                                                      ? { pathname: "/", id: 5 }
-                                                     : null);
+                                                     : undefined);
```
Comment (martinpitt) [inline pkg/storaged/block/format-dialog.jsx:293]: Does that have any observable effect?

**Review by martinpitt — APPROVED** [review]
Thanks! Pixel tests look correct.

**Comment by martinpitt** [comment]
@mvollmer [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19984-20240219-162518-1a4b8b2d-debian-stable-storage/log.html#99-1) is either a flake, or on debian it doesn't run twice properly. Retrying.

File: pkg/storaged/block/format-dialog.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/block/format-dialog.jsx:293]: Yes, get_fstab_config explicitly checks for `undefined` and would try to access `null.pathname`, for example. 

This was a bug in [storage: Improve Format dialog defaults](https://github.com/cockpit-project/cockpit/pull/19984/commits/721e1d6fe2cb7118b78e50b67d12d8dcd6bee45a), and I put it into the wrong commit... DO we care?

Comment (mvollmer) [reply pkg/storaged/block/format-dialog.jsx:293]: Ah, it's already merged. :-)

**Pull request merged**